darshan-runtime/lib/darshan-core.c

```diff
@@ -770,14 +802,23 @@ void darshan_core_shutdown(int write_log)
     for(i = 0; i < DARSHAN_KNOWN_MODULE_COUNT; i++)
         if(final_core->mod_array[i])
             final_core->mod_array[i]->mod_funcs.mod_cleanup_func();
+
```

Comment:
I don't know if it makes much difference, but if we wanted to we could move this a smidge earlier.  I think it would be safe immediately after darshan_log_finalize().

Reply:
Not sure which smidge you meant. Lines 805 to 810, unlink calls?
Do you mean moving them to after line 725?

Reply:
Oh yeah, I meant moving the final unlink (for the "dup" file) from it's current position to instead happen right after darshan_log_finalize().  The things in between that happen in between there and the current position don't have any impact on log output, so we can slightly reduce the window of time were we could leave a stray tmp file.

Reply:
It is done.